src/components/Navbar.tsx

```diff
@@ -1,9 +1,16 @@
 import { useQuery } from '@tanstack/react-query'
+import { Link } from '@tanstack/react-router'
 import { useEffect, useState } from 'react'
 import { getGenres } from '../api/categories'
-import { Link } from '@tanstack/react-router'
+import SearchBar from './SearchBar'
 
-function Navbar() {
+function Navbar({
+    search = false,
+    setSearchResult,
+}: {
+    search?: boolean
+    setSearchResult?: React.Dispatch<React.SetStateAction<string | undefined>>
```

Comment:
This feels not good. What is this for and why cant it be a handler function where the parent sets the state instead of a SetStateAction

Reply:
I fixed this by removing it and using zustand to store search query.